src/runtime/task/task_queue.cpp

```diff
@@ -54,44 +61,46 @@ void task_queue::enqueue_internal(task *task)
 {
     auto &sp = task->spec();
     auto throttle_mode = sp.rpc_request_throttling_mode;
-    if (throttle_mode != TM_NONE) {
+
+    if (throttle_mode != TM_NONE || FLAGS_global_max_task_queue_length > 0) {
```

Comment:
If the throttly_mode is TM_NONE, it should also be controlled by queue_length, this mode seems to be useless.